include/path.hpp

```diff
@@ -0,0 +1,141 @@
+#pragma once
+
+#include <iostream>
+#include <vector>
+#include <eigen3/Eigen/Dense>
+#include <eigen3/Eigen/Core>
+#include <glm/glm.hpp>
+
+
+// -------------------------------------------------------------------------------------------------------------------//
+
+// Misc definitions
+
+
+// Flags for boundary conditions
+// Since Eigen arrays don't work well with enum types, we just take integer flags
+typedef int bc_t;
+static const bc_t BC_T_START = 0, BC_ZERO = 0, BC_NATURAL = 1, BC_CLAMP = 2, BC_T_END = 3;
+
+
+// -------------------------------------------------------------------------------------------------------------------//
+
+// Matrices with a fixed number of columns, to be used either as vectors or as collection of vectors.
+
+
+// The `<type>Impl` templates are to be subclassed so the intended final `<type>` can overload constructors and
+// other stuff.
+template <size_t ndim> using
+BdryCondsImpl = Eigen::Array<bc_t, 2, ndim>;
+
+template <size_t ndim> using
+VectorImpl = Eigen::Matrix<float, 1, ndim>;
+
+template <size_t ndim> using
+VectorsImpl = Eigen::Matrix<float, Eigen::Dynamic, ndim>;
+
+
+template <size_t ndim>
+class Vector : public VectorImpl<ndim> {
+  public:
+    Vector() : VectorImpl<ndim>() {}
+    template<typename OtherDerived>
+    Vector(Eigen::MatrixBase<OtherDerived> const &other) : VectorImpl<ndim>(other) {}
+    Vector(glm::vec2 const &v) : VectorImpl<ndim>({v[0], v[1]}) { assert(ndim == 2); }
```

Comment:
Okay, good to know. But in fact Eigen already checks this, so my check was redundant.